### PR TITLE
Refactor partition/order/frame into a WindowNode.Specification

### PR DESCRIPTION
**Pull request by ebd2**
This is a precursor to future window function work.

### CODE REVIEW

**Comment by ebd2** [comment]
@maciejgrzybek

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java [hunk 1]
```diff
@@ -172,6 +141,58 @@ public int getPreSortedOrderPrefix()
...
+        {
+            return orderings;
+        }
+
+        @JsonProperty
+        public Frame getFrame()
+        {
+            return frame;
+        }
+    }
```
Comment (maciejgrzybek) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java:220]: Maybe you could consider adding `equals(Object)` and `hashCode()` methods?

**Comment by maciejgrzybek** [comment]
Looks good.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java (hunk 1 above)
Comment (ebd2) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/plan/WindowNode.java:220]: Added.

**Comment by ebd2** [comment]
Opening FB PR.

**Pull request closed without merging**